ivy_models/bert/bert.py

```diff
@@ -0,0 +1,353 @@
+        ("__v1__", "__1__"),
+        ("__v10__", "__2__"),
+        ("__v11__", "__3__"),
+    ]
+    key_map = key_map + [
+        (f"__v{i}", f".{j}") for i, j in zip(range(2, 10), range(4, 12))
+    ]
+    key_map = key_map + [
+        ("attention__dense", "attention.output.dense"),
+        ("attention__LayerNorm", "attention.output.LayerNorm"),
+    ]
+    key_map = key_map + [
+        ("ffd__dense1", "intermediate.dense"),
+        ("ffd__dense2", "output.dense"),
+        ("ffd__LayerNorm", "output.LayerNorm"),
+    ]
+    name = name.replace("__w", ".weight").replace("__b", ".bias")
+    name = (
+        name.replace("biasias", "bias")
+        .replace("weighteight", "weight")
+        .replace(".weightord", ".word")
+    )
+    for ref, new in key_map:
+        name = name.replace(ref, new)
+    name = name.replace("__", ".")
+    return name
+
+
+def get_idx_from_map(module_list, name):
+    names = ["v0", "v1", "v10", "v11"] + [
+        f"v{i}" for i in range(2, len(module_list) - 2)
+    ]
+    mapping = dict(zip(names, range(len(names))))
+    return mapping[name]
+
+
+def unflatten_set_module(
+    module,
+    flattened_name,
+    to_set,
+    split_on="__",
+):
+    splits = flattened_name.split(split_on)
+    cont = module
+    for idx, sp in enumerate(splits[:-1]):
+        cont = getattr(cont, sp)
+        if isinstance(cont, list):  # map the list structure to indices
+            mapped_idx = get_idx_from_map(cont, splits[idx + 1])
+            cont = cont[mapped_idx]
+            for s in splits[idx + 2 : -1]:
+                cont = getattr(cont, s)
+            break
+    cont = getattr(cont, "v")  # set the parameter variable to the wanted value
+    setattr(cont, splits[-1], to_set)
+
+
+def load_transformers_weights(
+    model, map_fn, model_name="bert-base-uncased", split_on="__"
+):
+    base = AutoModel.from_pretrained(model_name)
+    ref_weights = base.state_dict()
+    ref_weights = ivy.to_numpy(ivy.Container(ref_weights))
+    ivy.set_backend("torch")
+    old_mapping = copy.deepcopy(model.v)
+    param_names = old_mapping.cont_flatten_key_chains().keys()
+    mapping_list = map(lambda x: map_fn(x), param_names)
+    mapping = dict(zip(param_names, mapping_list))
+    ivy.previous_backend()
+    for old_name, ref_name in mapping.items():
+        to_set = ivy.asarray(ref_weights[ref_name])
+        unflatten_set_module(model, old_name, to_set, split_on)
+    return model
+
+
+def bert_base_uncased(pretrained=True):
+    # instantiate the hyperparameters same as bert
+    # set the dropout rate to 0.0 to avoid stochasticity in the output
+    config = BertConfig(
+        vocab_size=30522,
+        hidden_size=768,
+        num_hidden_layers=12,
+        num_attention_heads=12,
+        intermediate_size=3072,
+        hidden_act="gelu",
+        hidden_dropout=0.0,
+        attn_drop_rate=0.0,
+        max_position_embeddings=512,
+    )
+    model = BertModel(config, pooler_out=True)
+    if pretrained:
```

Comment:
once the weights helper refactor is completed, please ref resnet model on the code for replacing the weights